m3/reporter.go

```diff
@@ -694,7 +694,12 @@ func (r *reporter) reportInternalMetrics() {
 func (r *reporter) timeLoop() {
 	for !r.done.Load() {
 		r.now.Store(time.Now().UnixNano())
-		time.Sleep(_timeResolution)
+		select {
+		case <-time.After(_timeResolution):
```

Comment:
`time.After` allocates a new timer + channel on each call, from the docs:

> If efficiency is a concern, use NewTimer instead and call Timer.Stop if the timer is no longer needed.

Please use a timer instead.

Reply:
Replaced with NewTicker, we always wait for same duration.